### DISCUSSION TITLE
SSR navigation doesn't work in production with NextJS App Docker Image

### DISCUSSION THREAD

**Discussion by billypap1** [vercel/next.js#69845] [Help]
### Summary

I have deployed my NextJS App (App Router - v14.2.8) using Docker and when I navigate between SSR Routes it get stuck to infinitely loading state. For example, in a company page, I click a project item and it get stuck at loading, it doesn;t navaigate to the route.
I tested it locally  with npm run build & npm start and I didn't have any issue. 
This issue happens only in production deployed docker image. 
Any idea?

**Comment by Shriharsh-Deshmukh** [comment]
Hey **@billypap1** !

It looks like the issue is specific to your production Docker environment. Since everything works fine locally, a few things to check:

1. **Check Environment Variables:** Ensure your production `.env` file has the right variables and is accessible inside Docker.

2. **Network Config:** Sometimes, internal Docker networking can cause issues. Make sure Docker is correctly exposing necessary ports.

3. **Check Dockerfile:** Verify that your `Dockerfile` is optimized, ensuring dependencies are properly installed and no missing build steps.

4. **Next.js Cache:** Try clearing the `.next` cache before building the Docker image.

These should help resolve the infinite loading issue!
